core/infra/service/http_client.cpp

```diff
@@ -0,0 +1,169 @@
+Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <ctime>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <sys/uio.h>
+#include <sys/param.h>
+#include <unistd.h>
+
+#include <openssl/ssl.h>
+#include <openssl/crypto.h>
+#include <openssl/err.h>
+#include <openssl/rand.h>
+
+#include <json.hpp>
+
+#include <service/http_client.hpp>
+
+#include <util/logger.hpp>
+
+namespace http_client {
+
+    const std::string current() {
+        time_t t;
+        char timestr[256];
+        time(&t);
+        strftime(timestr, 255, "%a, %d %b %Y %H:%M:%S %Z", localtime(&t));
+        return std::string(timestr);
+    }
+
+    Request::Request(
+        std::string&& aMethod,
+        std::string&& aPath,
+        std::string&& abody
+    ):
+        method(move(aMethod)),
+        path(move(aPath)),
+        protocol("HTTP/1.1"),
+        body(move(abody))
+    {}
+
+    void Request::addHost(std::string host){
+        this->host = host;
+    }
+
+    void Request::addHeader(const std::string& key,std::string&& value){
+        headerset[key] = move(value);
+    }
+
+    void Request::addParams(const std::string& key,std::string&& value){
+        paramset[key] = move(value);
+    }
+
+    const std::string Request::dump(){
+        std::string res = method + " " + path + " " + protocol + "\n";
+        res += "Host: " + host + "\n";
+        for(const auto& it: headerset) {
+            res += it.first + ": " + it.second +"\n";
+        }
+        if(!body.empty()) {
+            res += "Content-Length: " + std::to_string(body.size()) + "\n";
+        }
+        res += "User-Agent: iroha\n";
+
+        res += "\n";
+        if(!body.empty()) {
+            res += body + "\n";
+        }
+        return res;
+    }
+
+    using nlohmann::json;
+
+    const int BUFFER_LENGTH = 2048;
+
+    int request(std::string dest, int port, Request req) {
+        req.addHost(dest);
+
+        struct hostent *hostent;
+        struct sockaddr_in server;
+
+        int fd;
+        char buffer[BUFFER_LENGTH];
+
+        hostent = gethostbyname(dest.c_str()); /* lookup IP */
+        if (hostent == nullptr) {
+            logger::error("HttpClient") <<  "Cannot resolve [" << dest << "]";
+            return 1;
+        }
+        bzero(&server, sizeof(server));
+        server.sin_family = AF_INET;
+        bcopy(hostent->h_addr, &server.sin_addr, hostent->h_length);
```

Comment:
http://stackoverflow.com/questions/18330673/bzero-bcopy-versus-memset-memcpy
`bcopy` is deprecated